Project/src/main/java/com/example/demo/exception/RestResponseEntityExceptionHandler.java

```diff
@@ -0,0 +1,36 @@
+package com.example.demo.exception;
+
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+
+@ControllerAdvice
+public class RestResponseEntityExceptionHandler
+        extends ResponseEntityExceptionHandler {
+
+    @ExceptionHandler(value
+            = { IllegalArgumentException.class, IllegalStateException.class })
+    protected ResponseEntity<Object> handleConflict(
+            RuntimeException ex, WebRequest request) {
+        String bodyOfResponse = "This should be application specific";
+        return handleExceptionInternal(ex, bodyOfResponse,
+                new HttpHeaders(), HttpStatus.CONFLICT, request);
+    }
+
+    /**
+     * In case there is a problem with the player details
+     * @param ex the exception
+     * @return a relevant response to client
+     */
+    @ExceptionHandler(PlayerNotFoundException.class)
+    protected ResponseEntity<Object> handlePlayerNotFound(PlayerNotFoundException ex, WebRequest request) {
+        String bodyOfResponse = "Player not found: " + ex.getMessage();
+        return new ResponseEntity<>(bodyOfResponse, new HttpHeaders(), HttpStatus.NOT_FOUND);
+    }
+}
```

Comment:
its good practice to have a "fallback" method for general exception
all exceptions extend Exception.class (the most generic java exception)

so if suddenly your app throws an exception you didnt think of, it will fall back to it

so make another method with 
@ExceptionHandler(Exception.class)